sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcSchemaIOProvider.java

```diff
@@ -67,6 +67,8 @@ public Schema configurationSchema() {
         .addNullableField("fetchSize", FieldType.INT16)
         .addNullableField("outputParallelization", FieldType.BOOLEAN)
         .addNullableField("autosharding", FieldType.BOOLEAN)
+        .addNullableField("maxConnections", FieldType.INT16)
```

Comment:
This is a breaking change for xlang schema transform, see #25062

in general we may need a better way to make this kind of change not breaking xlang

Reply:
I see, I think I find the cause. Here maxConnections is "FieldType.INT16" but in jdbc.py it is "typing.Optional[int]". This configuration gets encoded by ShortCoder then decode with VarIntCoder and data corruption happens.

Reply:
opened #26480